src/software/field_tests/field_test_fixture.py

```diff
@@ -373,84 +373,86 @@ def field_test_runner():
         debug_full_system=debug_full_sys,
         friendly_colour_yellow=args.run_yellow,
         should_restart_on_crash=False,
-    ) as friendly_fs, RobotCommunication(
+    ) as friendly_fs, Gamecontroller(
+        # we would be using conventional port if and only if we are playing in robocup.
+        supress_logs=(not args.show_gamecontroller_logs),
+        use_conventional_port=False,
+    ) as gamecontroller, RobotCommunication(
         current_proto_unix_io=friendly_proto_unix_io,
         multicast_channel=getRobotMulticastChannel(args.channel),
         interface=args.interface,
         estop_mode=estop_mode,
         estop_path=estop_path,
         enable_radio=args.enable_radio,
+        referee_port=Gamecontroller.get_referee_port_static(gamecontroller),
```

Comment:
for the future: the better way is to use the python ternary operator:

```suggestion
        referee_port=gamecontroller.get_referee_port() if gamecontroller else REFEREE_PORT,
```

why?
because if you're reading this code, there's an extra level of indirection where you have to peruse the implementation of the Gamecontroller module to figure out the `REFEREE_PORT` whereas it's very clear which referee port we bind to in the suggested code. We don't use `get_referee_port_static` enough around our codebase for this pattern to simplify the code

Reply:
I'm not going to block merging for this change since you'll lose your approvals but keep it in mind for the future

Reply:
This is actually life changing. 

I think a lot of the time I need to do something like that, but I just don't know that feature well enough to use it. we could get rid of this weird static method.

Reply:
Also, a lot of the time, I want initialize a variable when the being passed into the function is None.